pkg/controller/resourcegraphdefinition/controller_reconcile.go

```diff
@@ -78,13 +78,14 @@ func (r *ResourceGraphDefinitionReconciler) setupLabeler(rgd *v1alpha1.ResourceG
 
 // setupMicroController creates a new controller instance with the required configuration
 func (r *ResourceGraphDefinitionReconciler) setupMicroController(
+	ctx context.Context,
 	gvr schema.GroupVersionResource,
 	processedRGD *graph.Graph,
 	defaultSVCs map[string]string,
 	labeler metadata.Labeler,
 ) *instancectrl.Controller {
 
-	instanceLogger := r.rootLogger.WithName("controller." + gvr.Resource)
+	instanceLogger := ctrl.LoggerFrom(ctx).WithName(gvr.Resource)
```

Comment:
This causes the instance logger to share the same logger as the RGD reconciler, meaning that the instance reconciliation logs will contain unrelated RGD names/reconcile IDs. For example:

```
2025-03-27T21:52:13.884-0700	LEVEL(-2)	rgd-controller.dynamodbbackends	Using instance namespace	{"controller": "ResourceGraphDefinition", "controllerGroup": "kro.run", "controllerKind": "ResourceGraphDefinition", "namespace": "", "name": "dynamodbbackend", "reconcileID": "757a4daa-919a-4c40-b302-5e1a563c6358", "namespace": "kro", "name": "todoapp5", "resourceID": "iamPolicy", "namespace": "kro"}
```

Reply:
This one of the reasons why we setup two different loggers, one that is passed to RGD reconciler, one that is passed to the instance reconcilers.

Reply:
I had a think about this while I was making the change and I like that the named logger tells you where that thread come from, I think this sideeffect is a feature, not a bug :D

Reply:
>  will contain unrelated RGD names/reconcile IDs

oh, hmmm, let me think about this.

Reply:
sounds good to me